pkg/git/libgit2/managed/init.go

```diff
@@ -40,18 +40,8 @@ var (
 
 	debugLog logr.Logger
 	traceLog logr.Logger
-	enabled  bool
 )
 
-// Enabled defines whether the use of Managed Transport is enabled which
-// is only true if InitManagedTransport was called successfully at least
-// once.
-//
-// This is only affects git operations that uses libgit2 implementation.
-func Enabled() bool {
-	return enabled
-}
-
```

Comment:
Given how this has evolved through the last few releases, I think this should be kept, but renamed to `Initialized()` instead.

Without this we may not find a way to identify the corrupted state in which Managed Transport is enabled (feature gate) but not truly initialized.

Reply:
The purpose of removing this (and the related flag) was to avoid having multiple global states that refer to the same thing in general, since this can very easily cause hidden issues as the code grows. Making sure that there exists _only_ one way to check if managed transports are enabled, i.e. the feature gate, avoids this. To get information about the initialisation of the transports, the reconciler has been updated to contain that info, which it can easily pass down to the methods it calls using `CheckoutOptions`, which is more in line with the existing API.

Reply:
I'd prefer @aryan9600's approach here because it makes it easier to test different scenarios without trying to find ways to change the global state. The `ManagedTransportsRegistered` field in the gitrepo reconciler now allows easy testing of scenarios where managed transport initialization failed without affecting other tests.

Reply:
Initially, `managed.Enabled()` was introduced to work as the feature flag to switch the Managed Transport ON/OFF. It then gained a different meaning when feature gates were introduced, and unfortunately it was not renamed during that refactoring exercise. 

This now represents that the initialisation process was successfully executed once during the lifetime of the controller.

It is important to highlight that this has a complete different meaning to the `ManagedTransportsRegistered`, as the latter shows the intention to use Managed Transport. However, if those two values ever mismatch, that is an indication that we are operating at one of two possible corrupted states:

a) Managed Transport is wanted but was not initilised (e.g. errored during Init) 
b) Unmanaged Transport was wanted, but Managed Transport is being used.

This could be specially useful on tests to ensure we are testing what we believe/want to be testing.
Without keeping this here (and renaming for something more appropriate), we have no way (that I can think of) to detect such corrupt state.

Unless we decide to initialise the Managed Transport on `func init()`, which makes sense during Unmanaged transport deprecation, but probably not as part of the scope of this PR.

Reply:
@pjbgf I think there might me a misunderstanding. `ManagedTransportsRegistered` is a way for the reconciler to know whether `main()` was successful in registering/initializing our managed transports. If the `GitManagedTransport` feature gate is enabled, then the reconciler can use this information check if it should go ahead and reconcile the object (if `MangedTransportsRegistered=true`) or put the object in a stalled state (if `ManagedTransportsRegistered=false`).  The _intention_ to use managed transports is only derived via the feature gate, since that's a decision that resides on users.
@darkowlzz the opposite check isn't required since, we only attempt to register our managed transports when the feature gate was enabled, i.e. if Unmanaged Transport was wanted, it's impossible for our managed transports to be registered, and thus consequently `ManagedTransportsRegistered` is always `false`.